hack/app_sre_create_image_catalog.sh

```diff
@@ -26,9 +26,13 @@ git clone \
 # remove any versions more recent than deployed hash
 REMOVED_VERSIONS=""
 if [[ "$REMOVE_UNDEPLOYED" == true ]]; then
+    # this will need to be modified if at any point we move to a canary like deployment
+    # where not all environments are deployed with the same operator version
     DEPLOYED_HASH=$(
-        curl -s "https://gitlab.cee.redhat.com/service/app-interface/raw/master/data/services/osd-operators/cicd/saas/saas-${_OPERATOR_NAME}.yaml" | \
-            docker run --rm -i quay.io/app-sre/yq yq r - "resourceTemplates[*].targets(namespace.\$ref==/services/osd-operators/namespaces/hivep01ue1/${_OPERATOR_NAME}.yml).ref"
+        curl -s -H "Authorization: Basic $(echo -n ${APP_INTERFACE_USERNAME}:${APP_INTERFACE_PASSWORD} | base64)" \
+            -g "https://${APP_INTERFACE_BASE_URL}/graphql?query={saas_files:saas_files_v1{name,resourceTemplates{name,targets{namespace{environment{name,labels}},ref}}}}" | \
```

Comment:
@maorfr is there any way I could test the outcome of that curl command?

Reply:
You can [request creds from app-interface](https://gitlab.cee.redhat.com/service/app-interface#querying-the-app-interface) ([example](https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/10873)).

Reply:
When I tested this locally in pieces, I noticed that the output of the grapql query is sending a pretty large chunk of JSON across the wire (`< content-length: 162802`) and doing most of the winnowing in jq. Is there any way to pare that down on the server side? (I've never used graphql, so apologies if this is a silly question.)

Reply:
_responds to self_

Using `saas_files:saas_files_v1(path:"/services/osd-operators/cicd/saas/saas-${_OPERATOR_NAME}.yaml")` gets it down to `< content-length: 1846`. a) Relies on that file not moving -- but I guess we were relying on that originally anyway; and b) is slightly tricky to quote -- but that's easy to get around.

Reply:
I played with this a little bit, blowing out the pieces to (hopefully) make it more readable/understandable. This also includes what I mentioned above, to make the JSON payload ~100x smaller at the cost of hardcoding the saas file path.

```bash
    APP_INTERFACE_AUTH=$(echo -n "${APP_INTERFACE_USERNAME}:${APP_INTERFACE_PASSWORD}" | base64)
    GRAPHQL_QUERY=$(tr -d '\n ' <<EOF
{
  saas_files: saas_files_v1(path:"/services/osd-operators/cicd/saas/saas-${_OPERATOR_NAME}.yaml") {
    name,
    path,
    resourceTemplates {
      name,
      targets {
        namespace {
          environment {
            name,
            labels,
          }
        }
        ref
      }
    }
  }
}
EOF
)

    JQ_QUERY=$(tr -d '\n' <<EOF
   .data.saas_files[]
 | select(.name=="saas-${_OPERATOR_NAME}")
 | .resourceTemplates[].targets[]
 | select(.namespace.environment.labels
 | contains("\"type\":\"production\""))
 | .ref
EOF
)

    DEPLOYED_HASH=$(
        curl -s -H "Authorization: Basic ${APP_INTERFACE_AUTH}" \
            -g "https://${APP_INTERFACE_BASE_URL}/graphql?query=${GRAPHQL_QUERY}" | \
            jq -r "${JQ_QUERY}" | \
            uniq
    )
```

Reply:
Once https://github.com/app-sre/qontract-server/pull/93 and https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/11454 land, we should be able to do this as:

```shell
    APP_INTERFACE_AUTH=$(echo -n "${APP_INTERFACE_USERNAME}:${APP_INTERFACE_PASSWORD}" | base64)
    GRAPHQL_QUERY=$(tr -d '\n ' <<EOF
{
  saas_files: saas_files_v1(name:"saas-${_OPERATOR_NAME}") {
    resourceTemplates {
      targets {
        namespace {
          environment {
            labels,
          }
        }
        ref
      }
    }
  }
}
EOF
)

    JQ_QUERY=$(tr -d '\n' <<EOF
   .data.saas_files[]
 | .resourceTemplates[].targets[]
 | select(.namespace.environment.labels
 | contains("\"type\":\"production\""))
 | .ref
EOF
)

    DEPLOYED_HASH=$(
        curl -s -H "Authorization: Basic ${APP_INTERFACE_AUTH}" \
            -g "https://${APP_INTERFACE_BASE_URL}/graphql?query=${GRAPHQL_QUERY}" | \
            jq -r "${JQ_QUERY}" | \
            uniq
    )
```

or thereabouts, and get the double win of
- not having to specify the file path
- transferring only the file we need across the wire.

(@jmelis confirmed that there's already logic to enforce that the SaasFile_v1.name is globally unique, so we don't have to worry about that.)